tests/nosetests/pyanaconda_tests/module_payload_live_os_test.py

```diff
@@ -45,22 +45,25 @@ def setUp(self):
         self.live_os_module = LiveOSModule()
         self.live_os_interface = LiveOSInterface(self.live_os_module)
 
-        self.source_tests = SourceSharedTest(self,
-                                             payload=self.live_os_module,
-                                             payload_intf=self.live_os_interface)
+        self.shared_tests = PayloadSharedTest(self,
+                                              payload=self.live_os_module,
+                                              payload_intf=self.live_os_interface)
 
         self.callback = PropertiesChangedCallback()
         self.live_os_interface.PropertiesChanged.connect(self.callback)
 
     def _prepare_source(self):
-        return self.source_tests.prepare_source(SourceType.LIVE_OS_IMAGE)
+        return self.shared_tests.prepare_source(SourceType.LIVE_OS_IMAGE)
 
     def _prepare_and_use_source(self):
         source = self._prepare_source()
-        self.source_tests.set_sources([source])
+        self.shared_tests.set_sources([source])
 
         return source
 
+    def type_test(self):
+        self.shared_tests.check_type(PayloadType.LIVE_OS)
```

Comment:
I would also extend the test for `PayloadFactory`, because the type of the created payload should be the same as the requested type. It should be easy to check.

Reply:
Added. Thanks for a good idea.